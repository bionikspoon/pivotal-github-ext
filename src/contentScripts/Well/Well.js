import React from 'react'
import PropTypes from 'prop-types'
import cx from 'classnames'
import './Well.css'

export default function Well(props) {
  return (
    <div className={cx('Well__container', props.className)}>
      {props.children}
    </div>
  )
}
Well.defaultProps = {
  className: undefined,
}
Well.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired,
}
