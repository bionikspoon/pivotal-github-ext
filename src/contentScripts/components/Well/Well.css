.Well__container {
  margin: 15px 0;
  position: relative;
  background-color: #fff;
  border: 1px solid #ed7d1a;
  border-radius: 3px;

  &::before,
  &::after {
    position: absolute;
    top: 14px;
    right: 100%;
    left: -16px;
    display: block;
    width: 0;
    height: 0;
    pointer-events: none;
    content: ' ';
    border-color: transparent;
    border-style: solid solid outset;
  }

  &::before {
    border-width: 8px;
    border-right-color: #ed7d1a;
  }

  &::after {
    margin-top: 1px;
    margin-left: 2px;
    border-width: 7px;
    border-right-color: #fff;
  }
}
