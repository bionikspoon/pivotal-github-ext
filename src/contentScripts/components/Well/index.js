/* @flow */
export { default } from './Well'
