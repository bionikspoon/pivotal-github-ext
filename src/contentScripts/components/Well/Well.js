/* @flow */
import * as React from 'react'
import cx from 'classnames'
import './Well.css'

type Props = {
  className?: string,
  children: React.Node,
}

export default function Well(props: Props) {
  return (
    <div className={cx('Well__container', props.className)}>
      {props.children}
    </div>
  )
}
Well.defaultProps = {
  className: undefined,
}
