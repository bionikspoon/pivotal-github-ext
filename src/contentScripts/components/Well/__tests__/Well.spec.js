import React from 'react'
import { shallow } from 'enzyme'
import Well from '../Well'

describe('components/Well', () => {
  test('it renders correctly', () => {
    const props = {
      className: 'clearfix',
    }
    const wrapper = shallow(<Well {...props} />)

    expect(wrapper).toMatchSnapshot()
  })
})
