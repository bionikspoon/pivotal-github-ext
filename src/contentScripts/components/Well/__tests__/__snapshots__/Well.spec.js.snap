// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Well it renders correctly 1`] = `
<div
  className="Well__container clearfix"
/>
`;
